docs/source/tutorials/memory_optimizations.rst

```diff
@@ -0,0 +1,301 @@
+turn to stateless optimizers such as `stochastic gradient descent <https://pytorch.org/docs/stable/generated/torch.optim.SGD.html>`_
+without momentum, which don't require any additional memory usage, but will likely result in worse convergence during training.
+
+Can we find a middle ground here? Let's consider a technique which enables the use of "stateful" optimizers such as `AdamW <https://pytorch.org/docs/stable/generated/torch.optim.AdamW.html>`_
+without the memory overhead of gradient statistics, and without sacrificing their desirable convergence properties.
+How is this possible, you might ask? By *completely removing the buffer of gradients* which are stored by the optimizer during its ``step()``.
+
+To understand how this works, we encourage you to read through the relevant PyTorch tutorial on this concept:
+`How to save memory by fusing the optimizer step into the backward pass <https://pytorch.org/tutorials/intermediate/optimizer_step_in_backward_tutorial.html>`_.
+
+
+*Sounds great! How do I use it?*
+
+.. todo ref full finetune recipe doc
+
+In torchtune, you can enable this feature using the ``optimizer_in_bwd`` flag, which is currently only supported in our
+single-device full finetune recipe. This feature works best when gradient memory is particularly large
+i.e. when using a stateful optimizer with a model with a lot of parameters, and when you don't need to use
+:ref:`gradient accumulation <glossary_grad_accm>`.
+
+.. _glossary_peft:
+
+Parameter Efficient Fine-Tuning (PEFT)
+--------------------------------------
+
+.. _glossary_lora:
+
+Low Rank Adaptation (LoRA)
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+
+*What's going on here?*
+
+You can read our tutorial on :ref:`finetuning Llama2 with LoRA<lora_finetune_label>` to understand how LoRA works, and how to use it.
+Simply stated, LoRA greatly reduces the number of trainable parameters, thus saving significant gradient and optimizer
+memory during training.
+
+*Sounds great! How do I use it?*
+
+You can finetune using any of our recipes with the ``lora_`` prefix, e.g. :ref:`lora_finetune_single_device<lora_finetune_recipe_label>`. These recipes utilize
+LoRA-enabled model builders, which we support for all our models, and also use the ``lora_`` prefix, e.g.
+the :func:`torchtune.models.llama3.llama3` model has a corresponding :func:`torchtune.models.llama3.lora_llama3`.
+We aim to provide a comprehensive set of configurations to allow you to get started with training with LoRA quickly,
+just specify any config with ``_lora`` in its name, e.g:
+
+.. code-block:: bash
+
+  tune run lora_finetune_single_device --config llama3/8B_lora_single_device
+
+
+There are two sets of parameters to customize LoRA to suit your needs. Firstly, the parameters which control
+which linear layers LoRA should be applied to in the model:
+
+* ``lora_attn_modules: List[str]`` accepts a list of strings specifying which layers of the model to apply
```

Comment:
Its probably worth adding the tradeoff that finetuning more layers will require more memory and be slower, but will probably have higher performance, and that the bare minimum is probably only ["q_proj", "v_proj"]. I prefer how you described it, but for some inspiration, since this will probably be one of the most visited parameters: https://github.com/unslothai/unsloth/wiki#lora-parameters-encyclopedia